R/rdeephaven/R/table_handle_wrapper.R

```diff
@@ -1,147 +1,322 @@
-  },
-  signature = c("table_handle_instance")
-)
+    #' @description
+    #' Number of rows in the table referenced by this TableHandle.
+    #' @return The number of rows in the table.
+    nrow = function() {
+      return(self$.internal_rcpp_object$num_rows())
+    },
+    
+    ncol = function() {
+      return(self$.internal_rcpp_object$num_cols())
+    },
+    
+    dim = function() {
+      return(c(self$nrow(), self$ncol()))
+    },
+    
+    ### CONVERSION METHODS, ALSO IMPLEMENTED FUNCTIONALLY
 
-#' @export
-setMethod(
-  "is_static",
-  signature = c(table_handle_instance = "TableHandle"),
-  function(table_handle_instance) {
-    return(table_handle_instance@.internal_rcpp_object$is_static())
-  }
-)
+    #' @description
+    #' Imports the table referenced by this TableHandle into an Arrow RecordBatchStreamReader.
+    #' @return A RecordBatchStreamReader containing the data from the table referenced by this TableHandle.
+    as_record_batch_reader = function() {
+      ptr <- self$.internal_rcpp_object$get_arrow_array_stream_ptr()
+      rbsr <- RecordBatchStreamReader$import_from_c(ptr)
+      return(rbsr)
+    },
 
-#' @export
-setMethod(
-  "head",
-  signature = c(x = "TableHandle"),
-  function(x, n, ...) {
-    verify_positive_int("n", n, TRUE)
-    return(new("TableHandle", .internal_rcpp_object = x@.internal_rcpp_object$head(n)))
-  }
-)
+    #' @description
+    #' Imports the table referenced by this TableHandle into an Arrow Table.
+    #' @return A Table containing the data from the table referenced by this TableHandle.
+    as_arrow_table = function() {
+      rbsr <- self$as_record_batch_reader()
+      arrow_tbl <- rbsr$read_table()
+      return(arrow_tbl)
+    },
 
-#' @export
-setMethod(
-  "tail",
-  signature = c(x = "TableHandle"),
-  function(x, n, ...) {
-    verify_positive_int("n", n, TRUE)
-    return(new("TableHandle", .internal_rcpp_object = x@.internal_rcpp_object$tail(n)))
-  }
-)
+    #' @description
+    #' Imports the table referenced by this TableHandle into a dplyr Tibble.
+    #' @return A Tibble containing the data from the table referenced by this TableHandle.
+    as_tibble = function() {
+      rbsr <- self$as_record_batch_reader()
+      arrow_tbl <- rbsr$read_table()
+      return(as_tibble(arrow_tbl))
+    },
 
-#' @export
-setMethod(
-  "nrow",
-  signature = c(x = "TableHandle"),
-  function(x) {
-    return(x@.internal_rcpp_object$num_rows())
-  }
-)
+    #' @description
+    #' Imports the table referenced by this TableHandle into an R Data Frame.
+    #' @return A Data Frame containing the data from the table referenced by this TableHandle.
+    as_data_frame = function() {
+      arrow_tbl <- self$as_arrow_table()
+      return(as.data.frame(as.data.frame(arrow_tbl))) # TODO: for some reason as.data.frame on arrow table returns a tibble, not a data frame
+    },
+    
+    ### DEEPHAVEN TABLE OPERATIONS
+    
+    select = function(by = character()) {
```

Comment:
All of these functions need to get their arg names checked against the python names.  For example, `select` should not have `by` args.  Should check all functions, in case I just didn't notice elsewhere.

Reply:
I handled them all except for the argument names in the join functions, which I find to be far more informative than their Python analogs. `right_side`, `columns_to_match`, and `columns_to_add` in R, versus `table`, `on`, and `joins` in Python. I can change them if needed.

Reply:
Make them the same.